service/controller/resource/instance/instance_update_watchdog.go

```diff
@@ -3,11 +3,59 @@ package instance
 import (
 	"context"
 	"fmt"
+	"strconv"
+	"strings"
 	"time"
 
+	"github.com/Azure/go-autorest/autorest"
 	"github.com/giantswarm/microerror"
 )
 
+const (
+	vmssVMListHeaderName = "checkVMSSApiRateLimitThresholds"
+	headerKey3m          = "3m"
+	headerKey30m         = "30m"
+	max3m                = 190
+	max30m               = 900
+	threshold3m          = max3m * 0.5
+	threshold30m         = max30m * 0.5
+)
+
+func (r *Resource) checkVMSSApiRateLimitThresholds(response autorest.Response) (int64, int64) {
```

Comment:
```suggestion
func checkVMSSApiRateLimitThresholds(response autorest.Response) (int64, int64) {
```
Minor nitpick, but seems that this can be just a pure function.

Reply:
done